code/_core/obj/hud/inventory/interaction.dm

```diff
@@ -94,8 +111,21 @@
 			drop_item_from_inventory(get_turf(src))
 		return TRUE
 
-	if(grabbed_object && grabbed_object == object)
-		release_object(caller)
+	if(grabbed_object && grabbed_object == object && is_living(grabbed_object)) //We click on the hand that grabbed something
+		if(world.time <= grab_time+DECISECONDS_TO_TICKS(20)) //Prevents insta agressive-grab
+			caller.to_chat(span("warning","Too soon to tighting your grip on [object]! Need to wait another [TICKS_TO_SECONDS(grab_time+DECISECONDS_TO_TICKS(20) - world.time)] seconds!"))
+			return TRUE
+		caller.to_chat(span("warning","You tighten your grip on [object]!"))
+		grab_level = 2 //Agressive grab
+		grab_time = world.time //We grabbed now. Used in agressive grab
+
+		var/mob/living/A = owner //How can something not alive grab anything?..
+		A.add_status_effect(SLOW,30,30) //I don't know any better way..
+		var/mob/living/V = grabbed_object //Checked above
+		V.add_status_effect(REST,30,30)
+		V.add_status_effect(SLOW,30,30)
```

Comment:
Just to let you know, all of theses status effects will only last for 3 seconds. Is that intended?

Reply:
Initially, I wanted to make it permanent (or until drop), but then looking at the experience from other servers, I thought that permastun would be far from the most interesting feature. If you managed to grab something, you have upper hand for some time, but it shouldn't be the end.